ApplensBackend/Services/FreshChatClientService/FreshChatClientService.cs

```
@@ -0,0 +1,410 @@
+            _resourceConversationCache =  new ConcurrentDictionary<string, ResourceCacheEntry>();
+    }
+
+        /// <summary>
+        /// Add or update agent information in agent cache.
+        /// </summary>
+        /// <param name="agentId">Identifier for the agent.</param>
+        /// <param name="value">Details of the Agent. See <see cref="AgentDetails"/>.</param>
+        private void AddOrUpdateAgent(string agentId, AgentDetails value)
+        {
+            if (!string.IsNullOrWhiteSpace(agentId))
+            {
+                AgentCacheEntry newEntry = new AgentCacheEntry();
+                newEntry.agent = value;
+                newEntry.timeStamp = DateTime.Now;
+                _agentCache.AddOrUpdate(agentId.ToLower(), newEntry, (existingKey, oldValue) => newEntry);
+
+                if (_agentCache.Count > maxCacheEntries)
+                {
+                    TryRemoveAgent(_agentCache.OrderBy(o => o.Value.timeStamp).FirstOrDefault().Key);
+                }
+            }
+        }
+
+        private bool TryRemoveAgent(string agentId)
+        {
+            AgentCacheEntry ignored;
+            return _agentCache.TryRemove(agentId.ToLower(), out ignored);
+        }
+
+        /// <inheritdoc/>
+        public async Task<AgentDetails> GetAgentDetails(string agentId)
+        {
+            AgentCacheEntry cacheEntry;
+            bool result = _agentCache.TryGetValue(agentId.ToLower(), out cacheEntry);
+            if (result)
+            {
+                return cacheEntry.agent;
+            }
+            else
+            {
+                AgentDetails currAgent = new AgentDetails();
+                string uri = $"/agents/{agentId}";
+                try
+                {
+                    HttpResponseMessage response = await MakeGetCallToFreshChat(uri);
+                    string responseContent = await response.Content.ReadAsStringAsync();
+                    var jsonResponse = JToken.Parse(responseContent);
```

this could throw the exception in case http call returns a 500 or non-success code. 
Do you want  to handle that exception or let it throw. (if later, maybe you can put a try catch and throw the responseContent as part of the exception)